// nativescript-toasty/toasty.ios.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var color_1 = require("tns-core-modules/color");
var platform_1 = require("tns-core-modules/platform");
var enums_1 = require("tns-core-modules/ui/enums");
var frameModule = require("tns-core-modules/ui/frame");
var toast_common_1 = require("./toast.common");
__export(require("./toast.common"));
var Toasty = (function () {
    function Toasty(opts) {
        this._csToaststyle = CSToastStyle.alloc().initWithDefaultStyle();
        this._text = opts.text;
        this._duration = opts.duration;
        this._position = opts.position;
        this._textColor = opts.textColor;
        this._backgroundColor = opts.backgroundColor;
        this._iOSOpts = opts.ios || {};
        this.setToastDuration(this._duration)
            .setToastPosition(this._position)
            .setTextColor(this._textColor)
            .setBackgroundColor(this._backgroundColor);
        if (this._iOSOpts.displayShadow) {
            this._csToaststyle.displayShadow = this._iOSOpts.displayShadow;
            if (this._iOSOpts.shadowColor) {
                if (typeof this._iOSOpts.shadowColor === 'string') {
                    this._csToaststyle.shadowColor = new color_1.Color(this._iOSOpts.shadowColor).ios;
                }
                else {
                    this._csToaststyle.shadowColor = this._iOSOpts.shadowColor.ios;
                }
            }
        }
        if (this._iOSOpts.cornerRadius) {
            this._csToaststyle.cornerRadius = this._iOSOpts.cornerRadius;
        }
        if (this._iOSOpts.messageNumberOfLines) {
            this._csToaststyle.messageNumberOfLines = this._iOSOpts.messageNumberOfLines;
        }
        CSToastManager.setSharedStyle(this._csToaststyle);
        return this;
    }
    Object.defineProperty(Toasty.prototype, "position", {
        get: function () {
            return this._position;
        },
        set: function (value) {
            if (value) {
                this._position = value;
                this.setToastPosition(value);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Toasty.prototype, "duration", {
        get: function () {
            return this._duration;
        },
        set: function (value) {
            if (value) {
                this._duration = value;
                this.setToastDuration(value);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Toasty.prototype, "textColor", {
        get: function () {
            return this._textColor;
        },
        set: function (value) {
            if (value) {
                this._textColor = value;
                this.setTextColor(value);
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(Toasty.prototype, "backgroundColor", {
        get: function () {
            return this._backgroundColor;
        },
        set: function (value) {
            if (value) {
                this._backgroundColor = value;
                this.setBackgroundColor(value);
            }
        },
        enumerable: true,
        configurable: true
    });
    Toasty.prototype.show = function () {
        if (!this._text) {
            throw new Error('Text is not set');
        }
        else {
            this._getView().makeToast(this._text);
        }
    };
    Toasty.prototype.cancel = function () {
        this._getView().hideToasts();
    };
    Toasty.prototype.setTextColor = function (value) {
        if (value) {
            this._textColor = value;
            if (typeof value === 'string') {
                var nativeColor = new color_1.Color(value).ios;
                this._csToaststyle.messageColor = nativeColor;
            }
            else {
                this._csToaststyle.messageColor = value.ios;
            }
            CSToastManager.setSharedStyle(this._csToaststyle);
        }
        return this;
    };
    Toasty.prototype.setBackgroundColor = function (value) {
        if (value) {
            this._backgroundColor = value;
            if (typeof value === 'string') {
                var nativeColor = new color_1.Color(value).ios;
                this._csToaststyle.backgroundColor = nativeColor;
            }
            else {
                this._csToaststyle.backgroundColor = value.ios;
            }
            CSToastManager.setSharedStyle(this._csToaststyle);
        }
        return this;
    };
    Toasty.prototype.setToastPosition = function (value) {
        switch (value) {
            case toast_common_1.ToastPosition.TOP:
                CSToastManager.setDefaultPosition(CSToastPositionTop);
                break;
            case toast_common_1.ToastPosition.CENTER:
                CSToastManager.setDefaultPosition(CSToastPositionCenter);
                break;
            case toast_common_1.ToastPosition.BOTTOM:
                CSToastManager.setDefaultPosition(CSToastPositionBottom);
                break;
            default:
                break;
        }
        return this;
    };
    Toasty.prototype.setToastDuration = function (value) {
        switch (value) {
            case toast_common_1.ToastDuration.SHORT:
                CSToastManager.setDefaultDuration(2.0);
                break;
            case toast_common_1.ToastDuration.LONG:
                CSToastManager.setDefaultDuration(4.0);
                break;
            default:
                CSToastManager.setDefaultDuration(2.0);
                break;
        }
        return this;
    };
    Toasty.prototype._getView = function () {
        if (!frameModule.topmost()) {
            throw new Error('There is no topmost');
        }
        else {
            if (this._iOSOpts && this._iOSOpts.anchorView) {
                return this._iOSOpts.anchorView;
            }
            else {
                var viewController = frameModule.topmost()
                    .viewController;
                if (viewController.presentedViewController) {
                    if (platform_1.device.deviceType !== enums_1.DeviceType.Tablet) {
                        while (viewController.presentedViewController) {
                            viewController = viewController.presentedViewController;
                        }
                    }
                }
                return viewController.view.window;
            }
        }
    };
    return Toasty;
}());
exports.Toasty = Toasty;
